src/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/FileStreamReadLightUp.cs

```diff
@@ -156,7 +156,7 @@ IntPtr overlapped
             );
 
             // older Windows systems:
-            [DllImport(@"kernel32.dll", EntryPoint = "ReadFile", ExactSpelling = true, SetLastError = true)]
+            [DllImport(@"api-ms-win-core-file-l1-1-0.dll", EntryPoint = "ReadFile", ExactSpelling = true, SetLastError = true)]
```

Comment:
> Note that both Pinvoke's are incorrect

The P/Invokes are guarded by light-up. If both fail to work we fall back to a slower code path. So while I'm not saying that there isn't a genuine WACK issue, I would like to clarify that there is no x-plat (meaning non-Windows) bug here.

>The right fix would be to separate the Desktop implementation and make this library windows specific (or remove windows Pinvokes from the x-plat implementation)

I would really like to avoid this if at all possible. A lib like S.R.M shouldn't need to build separately for Windows and Unix when it is so far removed from the OS in its function.

> This API should really be provided by System.IO.FileSystem.

Yes. I filed #4786 a while ago. Unfortunately, no addition to the BCL can handle the optimization if we want it to continue to kick in on downlevel .NET Framework and Windows.